djstripe/models.py

```diff
@@ -52,6 +51,40 @@
 djstripe_settings.set_stripe_api_version()
 
 
+class PaymentMethod(models.Model):
+    """
+    An internal model that abstracts the legacy Card and BankAccount
+    objects with Source objects.
+
+    Contains two fields: `id` and `type`:
+    - `id` is the id of the Stripe object.
+    - `type` can be `card`, `bank_account` or `source`.
+    """
+    id = CharField(max_length=255, primary_key=True)
+    type = CharField(max_length=12, db_index=True)
+
+    @classmethod
+    def _get_or_create_source(cls, data, source_type):
+        # TODO other source types
+        if source_type == SourceType.card:
+            Card._get_or_create_from_stripe_object(data)
+
+        return cls.objects.get_or_create(id=data["id"], defaults={"type": source_type})
+
+    @property
+    def stripe_id(self):
+        # Deprecated (transitional)
+        return self.id
+
+    @property
+    def object_model(self):
+        # TODO other source types
+        return Card
+
+    def get_object(self):
+        return self.object_model.objects.get(stripe_id=self.id)
```

Comment:
Slightly concerned about the double de-reference necessary to get the underlying, but only slightly. :-)

Reply:
Might be worth adding a convenience accessor at least:

```
@cached_property
def source(self):
    return self.get_object()
```